tests/integration/security/authorization_test.go

```diff
@@ -1334,28 +1334,34 @@ func TestAuthorization_Custom(t *testing.T) {
 			// Path "/health" is not protected and is accessible to public.
 			cases := []rbacUtil.TestCase{
 				// workload b is using an ext-authz service in its own pod of HTTP API.
-				newTestCase(b, "/custom", "allow", true),
-				newTestCase(b, "/custom", "deny", false),
-				newTestCase(b, "/health", "allow", true),
-				newTestCase(b, "/health", "deny", true),
+				newTestCase(a, b, "/custom", "allow", true, scheme.HTTP),
+				newTestCase(a, b, "/custom", "deny", false, scheme.HTTP),
+				newTestCase(a, b, "/health", "allow", true, scheme.HTTP),
+				newTestCase(a, b, "/health", "deny", true, scheme.HTTP),
 
 				// workload c is using an ext-authz service in its own pod of gRPC API.
-				newTestCase(c, "/custom", "allow", true),
-				newTestCase(c, "/custom", "deny", false),
-				newTestCase(c, "/health", "allow", true),
-				newTestCase(c, "/health", "deny", true),
+				newTestCase(a, c, "/custom", "allow", true, scheme.HTTP),
+				newTestCase(a, c, "/custom", "deny", false, scheme.HTTP),
+				newTestCase(a, c, "/health", "allow", true, scheme.HTTP),
+				newTestCase(a, c, "/health", "deny", true, scheme.HTTP),
 
 				// workload d is using an local ext-authz service in the same pod as the application of HTTP API.
-				newTestCase(d, "/custom", "allow", true),
-				newTestCase(d, "/custom", "deny", false),
-				newTestCase(d, "/health", "allow", true),
-				newTestCase(d, "/health", "deny", true),
+				newTestCase(a, d, "/custom", "allow", true, scheme.HTTP),
+				newTestCase(a, d, "/custom", "deny", false, scheme.HTTP),
+				newTestCase(a, d, "/health", "allow", true, scheme.HTTP),
+				newTestCase(a, d, "/health", "deny", true, scheme.HTTP),
 
 				// workload e is using an local ext-authz service in the same pod as the application of gRPC API.
-				newTestCase(e, "/custom", "allow", true),
-				newTestCase(e, "/custom", "deny", false),
-				newTestCase(e, "/health", "allow", true),
-				newTestCase(e, "/health", "deny", true),
+				newTestCase(a, e, "/custom", "allow", true, scheme.HTTP),
+				newTestCase(a, e, "/custom", "deny", false, scheme.HTTP),
+				newTestCase(a, e, "/health", "allow", true, scheme.HTTP),
+				newTestCase(a, e, "/health", "deny", true, scheme.HTTP),
+
+				// workload b is using an ext-authz service in its own pod of TCP API.
+				newTestCase(a, b, "/custom-tcp", "", true, scheme.TCP),
```

Comment:
this is not right, to test tcp, send requests to 2 different ports, one port is specified in the authz for ext-authz. and do not use path, it does not make sense for tcp.